src/anonymousUserTracking/anonymousUserEventManager.ts

```diff
@@ -149,7 +149,7 @@ export class AnonymousUserEventManager {
     this.storeEventListToLocalStorage(newDataObject, false);
   }
 
-  private async checkCriteriaCompletion() {
+  private checkCriteriaCompletion() {
```

Comment:
can we type these methods out? would be helpful to know all of their return types.